test/export/test_export.py

```diff
@@ -226,9 +227,11 @@ def forward(self, x):
         inp = (torch.tensor([3]),)
         ep = export(ConflictingConstraints(), inp)
 
-        with self.assertRaisesRegex(
-            RuntimeError, r"Invalid value range for 3 between \[4, 5\]"
-        ):
+        if is_non_strict_test(self._testMethodName):
```

Comment:
Hmm why does it change?

Reply:
Original (main):
```
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, x: "i64[1]"):
             # File: /data/users/pianpwk/copy/pytorch/test/export/test_export.py:220 in forward, code: b = x.item()
            item: "Sym(u0)" = torch.ops.aten.item.default(x);  x = None
            
             # 
            sym_constrain_range_for_size_default = torch.ops.aten.sym_constrain_range_for_size.default(item)
            
            # No stacktrace found for following nodes
            sym_constrain_range = torch.ops.aten.sym_constrain_range.default(item, min = 4, max = 5)
            ge: "Sym(u0 >= 0)" = item >= 0
            _assert_scalar = torch.ops.aten._assert_scalar.default(ge, "Runtime assertion failed for expression 0 <= u0 on node 'ge_1'");  ge = None
            ge_1: "Sym(u0 >= 4)" = item >= 4
            _assert_scalar_1 = torch.ops.aten._assert_scalar.default(ge_1, "Runtime assertion failed for expression 4 <= u0 on node 'ge_2'");  ge_1 = None
            le: "Sym(u0 <= 5)" = item <= 5
            _assert_scalar_2 = torch.ops.aten._assert_scalar.default(le, "Runtime assertion failed for expression u0 <= 5 on node 'le_1'");  le = None
            
             # File: /data/users/pianpwk/copy/pytorch/test/export/test_export.py:224 in forward, code: return torch.full((b, 1), 1)
            full: "i64[u0, 1]" = torch.ops.aten.full.default([item, 1], 1, device = device(type='cpu'), pin_memory = False);  item = None
            return (full,)
```

Current (strict-mode): the assert_async calls are added separately, and should be removed and the pass deprecated soon: https://github.com/pytorch/pytorch/pull/125948. For non-strict we don't run this pass.

```
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, x: "i64[1]"):
             # File: /data/users/pianpwk/pytorch/test/export/test_export.py:221 in forward, code: b = x.item()
            item: "Sym(u0)" = torch.ops.aten.item.default(x);  x = None
            
             # 
            ge: "Sym(u0 >= 4)" = item >= 4
            scalar_tensor_default: "f32[]" = torch.ops.aten.scalar_tensor.default(ge);  ge = None
            _assert_async_msg = torch.ops.aten._assert_async.msg(scalar_tensor_default, 'item is outside of inline constraint [4, 5].');  scalar_tensor_default = None
            le: "Sym(u0 <= 5)" = item <= 5
            scalar_tensor_default_1: "f32[]" = torch.ops.aten.scalar_tensor.default(le);  le = None
            _assert_async_msg_1 = torch.ops.aten._assert_async.msg(scalar_tensor_default_1, 'item is outside of inline constraint [4, 5].');  scalar_tensor_default_1 = None
            
             # File: /data/users/pianpwk/pytorch/torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:158 in call, code: node.meta["stack_trace"] = "".join(traceback.format_stack(limit=1))
            sym_constrain_range_for_size = torch.ops.aten.sym_constrain_range_for_size.default(item, min = 4, max = 5)
            
             # File: /data/users/pianpwk/pytorch/test/export/test_export.py:225 in forward, code: return torch.full((b, 1), 1)
            full: "i64[u0, 1]" = torch.ops.aten.full.default([item, 1], 1, device = device(type='cpu'), pin_memory = False);  item = None
            return (full,)
```

Current (non-strict), ideal:
```
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, x: "i64[1]"):
             # File: /data/users/pianpwk/pytorch/test/export/test_export.py:221 in forward, code: b = x.item()
            item: "Sym(u1)" = torch.ops.aten.item.default(x);  x = None
            
             # 
            sym_constrain_range_for_size_default = torch.ops.aten.sym_constrain_range_for_size.default(item, min = 4, max = 5)
            
             # File: /data/users/pianpwk/pytorch/test/export/test_export.py:225 in forward, code: return torch.full((b, 1), 1)
            full: "i64[u1, 1]" = torch.ops.aten.full.default([item, 1], 1, device = device(type='cpu'), pin_memory = False);  item = None
            return (full,)
```

Reply:
Others are a similar story, should be unified b/w strict & non-strict once the above PR is merged